specs/OAuth/OAuth2Manager.md

```diff
@@ -0,0 +1,956 @@
+| HttpFailure | The HTTP POST request failed. See the 'ErrorCode' property for more details as to why. |
+| InvalidResponse | The server responded, but its response was improperly formatted. This could be that the server did not send the response as JSON, the response JSON string was improperly formatted, or the response JSON contained unexpected object types (e.g. a number when a string is expected, etc.). |
+
+## TokenFailure class
+
+It's a class that provides the failure object of a token request.
+
+## TokenFailure Properties
+
+| Name | Description | Type |
+|-|-|-|
+| Kind | Specifies the kind of token failure. | TokenFailureKind |
+| Error | Derived from the "error" parameter of the error response. | String |
+| ErrorDescription | Derived from the "error_description" parameter of the error response. | String |
+| ErrorUri | Derived from the "error_uri" parameter of the error response. | Windows.Foundation.Uri |
+| ErrorCode | The HTTP error code that was returned. | Windows.Web.Http.HttpStatusCode |
+| AdditionalParams | Additional parameters set by the authorization server in the response URI. | Windows.Foundation.Collections.IMapView<String, String> |
+
+
+# API Details
+
+> Note: all of this new WinAppSDK API is to support 
+  OAuth 2.0 launch using browser launch.
+ 
+```c++ (but really MIDL3)
+namespace Microsoft.Security.Authentication.OAuth
+{
+    [contractversion(1)]
+    apicontract OAuthContract {};
+
+    [contract(OAuthContract, 1)]
+    runtimeclass ClientAuthentication
+    {
+        ClientAuthentication();
+        ClientAuthentication(Windows.Web.Http.Headers.HttpCredentialsHeaderValue authorization);
+
+        static ClientAuthentication CreateForBasicAuthorization(String clientId, String clientSecret);
+
+        // Specifies the 'Authorization' header of the HTTP POST request when requesting a token
+        Windows.Web.Http.Headers.HttpCredentialsHeaderValue Authorization { get; set; };
+
+        // Specifies the 'Proxy-Authorization' header of the HTTP POST request when requesting a token
+        Windows.Web.Http.Headers.HttpCredentialsHeaderValue ProxyAuthorization { get; set; };
+
+        // Specifies additional header values of the HTTP POST request when requesting a token
+        Windows.Foundation.Collections.IMap<String, String> AdditionalHeaders { get; };
+    }
+
+    [contract(OAuthContract, 1)]
+    static runtimeclass OAuth2Manager
+    {
+        // Initiates an authorization request in the user's default browser as described by RFC 6749 section 3.1. The
+        // returned 'IAsyncOperation' will remain in the 'Started' state until it is either cancelled or completed by a
+        // call to 'CompleteAuthRequest'. This performs authorization of response_type="token".
+        static Windows.Foundation.IAsyncOperation<AuthRequestResult> InitiateAuthRequestAsync(Microsoft.UI.WindowId parentWindowId,
```

Comment:
This kind of operation we'd name `StartThingAsync` (if there's no user interaction) or `RequestThingAsync` (because it makes a _request_ of information.) So I think these would all be like `RequestAuthAsync`, like:

* `RequestAuthAsync(windowId, completeAuthEndpoint)`
* `RequestAuthAsync(windowId, completeAuthEndpoint, redirectUri)`
* `RequestAuthWithParamsAsync(windowId, authEndpoint, params)`

No change required to get going on preview.